swift/src/Ice/ObjectAdapterI.swift

```diff
@@ -263,19 +263,21 @@ class ObjectAdapterI: LocalObject<ICEObjectAdapter>, ObjectAdapter, ICEDispatchA
 
         let request = IncomingRequest(current: current, inputStream: istr)
 
-        dispatchPipeline.dispatch(request).map { response in
+        do {
```

Comment:
Is function is called from ObC. The memory held by the input stream needs to survive the dispatch as we don't know when decoding will happen.

Reply:
Ideally, this memory would only survive the decoding - not the dispatch. That's what we do in IceRPC+Slice with the help of System.IO.Pipelines.

In Ice C++, the situation is a bit different because there is no real "async" dispatch:
https://github.com/zeroc-ice/ice/blob/d3542d6e1e2474e4b29dfc796e2c9dd86c23a4ad/cpp/include/Ice/Object.h#L35

The C++ IncomingRequest (and the InputStream payload) is allocated on the stack of the dispatch thread, and it's not supposed to be passed across threads (that's why it's a reference, not a copy). Its payload should be decoded only by the dispatch thread. The whole object is released when the synchronous part of the dispatch completes.

For async dispatches, we copy Current from IncomingRequest when we build the AsyncResponseHandler:
https://github.com/zeroc-ice/ice/blob/d3542d6e1e2474e4b29dfc796e2c9dd86c23a4ad/cpp/include/Ice/AsyncResponseHandler.h#L23

This way, it survives until the response is provided asynchronously.

I am just explaining how this works in Ice C++... not suggesting we should do the same in Swift.